src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -1681,6 +1681,19 @@ namespace shogun
 			return infer_backend(A)->transpose_matrix(A);
 		}
 
+		/**
+		 * Method that computes the trace of \f$AB\f$ as \f$sum(A.*B')$\f
+		 *
+		 * @param A The matrix A
+		 * @param B The matrix B
+		 * @return The trace of the product of A and B
+		 */
+		template <typename T>
+		T trace_dot(const SGMatrix<T>& A, const SGMatrix<T>& B)
+		{
+			return sum(element_prod(A, transpose_matrix(B)));
```

Comment:
is transpose matrix changing the memory (it shouldnt)

Also, the API seems inconsistent, for some methods we pass flags for whether matrices should be transposed. And for others, this flag doesnt exist and we have to transpose matrices from the outside ?

Reply:
transpose matrix creates a new matrix.  
The API is indeed inconsistent, maybe we should add transpose flags to element_prod ?

Reply:
yep!

Reply:
@karlnapf I made another pr #4183 , with which I could make it clear here